File: src/main/java/com/github/weewar/mapviewer/startup/PopulateInMemoryDAO.java
package com.github.weewar.mapviewer.startup;

import com.github.weewar.mapviewer.dao.memory.InMemoryWeewarMapDAO;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

@Component
public class PopulateInMemoryDAO {
    private final Logger logger = LoggerFactory.getLogger(InMemoryWeewarMapDAO.class);

    @Autowired(required = false)
    public void onStartup(InMemoryWeewarMapDAO inMemoryMapDAO) {
        logger.info("Populating in-memory DAO.");
        inMemoryMapDAO.populate();
    }
}

File: src/main/kotlin/com/github/weewar/mapviewer/startup/PopulateInMemoryDAO.kt
package com.github.weewar.mapviewer.startup

import com.github.weewar.mapviewer.dao.memory.InMemoryWeewarMapDAO
import org.slf4j.LoggerFactory.getLogger
import org.springframework.boot.context.event.ApplicationReadyEvent
import org.springframework.context.event.EventListener
import org.springframework.stereotype.Component

@Component
class PopulateInMemoryDAO(val inMemoryDao: InMemoryWeewarMapDAO) {
    private val logger = getLogger(PopulateInMemoryDAO::class.java)

    @EventListener(ApplicationReadyEvent::class)
    fun onStartup() {
        logger.info("Populating in-memory repository")
        inMemoryDao.populate()
    }
}